base/set.jl

```diff
@@ -167,15 +167,37 @@ julia> unique(x -> x^2, [1, -1, 3, -3, 4])
 """
 function unique(f, C)
     out = Vector{eltype(C)}()
-    seen = Set()
-    for x in C
-        y = f(x)
-        if !in(y, seen)
-            push!(seen, y)
+
+    s = iterate(C)
+    if s === nothing
+        return out
+    end
+    (x, i) = s
+    seen = Set{typeof(x)}()
+    push!(seen, x)
+    push!(out, x)
+
+    return _unique!(out, f, C, seen, i)
+end
+
+function _unique!(out::AbstractVector, f, C, seen::Set, i)
```

Comment:
Preferred argument order is function, before mutated output.
(It is, IIRC the only thing that is).

I see no reason not to follow that convention for internal methods.